### DISCUSSION TITLE
Automatically enable light or dark variants based on terminal colors.

### DISCUSSION THREAD

**Discussion by wtwingate** [catppuccin/nvim#629] [Q&A] [answered]
I apologize in advance if I've missed something obvious, but I haven't been able to get this to work.

For context, I'm using Neovim 0.9.4 on Mac with iTerm2.

I want Neovim to set my colorscheme to either catppuccin-mocha when using a dark terminal theme, or catppuccin-latte when using a light terminal theme. Based on the default options for catppuccin, this seems like it should work "out of the box," but I can't get it to work. 

In my init.lua, I set catppuccin as the default color scheme with `vim.cmd.colorscheme("catppuccin")`. However, regardless of what terminal colors are used, `set background?` always returns `background=dark` when starting Neovim with catppuccin. 

Oddly enough, I don't encounter this problem with other color schemes. For example, the background color is correctly determined when using tokyonight, and the appropriate light or dark variant is enabled automatically.

Does anyone know of a solution to this problem?

**Comment by mrtnvgr** [comment]
Have you tried to change [that line](https://github.com/wtwingate/dotfiles/blob/main/nvim%2Flua%2Foptions.lua#L22) to `catppuccin`? Also, you can try running `:CatppuccinCompile`.

**Reply by wtwingate** [reply]
Yes I have. When set to `catppuccin` it always  loads as `catppuccin-mocha`.

**Reply by wtwingate** [reply]
Also, I ran `:CatppuccinCompile` but the behavior is unchanged.

Based on what I've seen, the cause of the problem seems to be that when `vim.cmd.colorscheme("catppuccin")` is included in init.lua, Neovim fails to automatically set the background color based on the terminal color scheme, and so it defaults to `background=dark`.

Once Neovim is loaded, `set background=light` and `set background=dark` causes catppuccin to switch between `latte` and `mocha` like it's supposed to do. So that's not the problem.

Likewise, if `vim.cmd.colorscheme("catppuccin")` is excluded from init.lua, then the background color is automatically detected and set by Neovim  like it's supposed to. Afterwards, if catppuccin is loaded manually with `colorscheme catppuccin` it works as expected.

**Comment by the-spanish-guy** [comment]
u need set `vim.o.background = "light"` at the end of catppuccin config file

```lua
return {
  "catppuccin/nvim",
  name = "catppuccin",
  priority = 1000,
  config = function()
    require("catppuccin").setup({
      flavour = "mocha",
      background = {
        light = "latte",
        dark = "mocha",
      },
    })
    vim.cmd.colorscheme("catppuccin")
    vim.o.background = "light"
  end,
}

```

**Reply by Minjares** [reply]
That just makes the theme light for me everytime.

**Comment by rewhile** [comment, marked as answer]
This should get fixed in the latest commit

Refer to https://github.com/catppuccin/nvim/issues/682 for more infos

**Reply by wtwingate** [reply]
It is, thank you so much!
